### PR TITLE
Speed up tests 608

### PR DESCRIPTION
**Pull request by phlptt**
* removal of exporter dependency
* go straight to application for speed
* clear_down function to move all cases from new cases to application bin (note this is not currently called its jsut there to be used)

### CODE REVIEW

File: ui_automation_tests/helpers/seed_data.py [hunk 1]
```diff
@@ -205,13 +205,31 @@ def add_queue(self, queue_name):
...
     def assign_case_to_queue(self, case_id=None, queue_id=None):
         self.log("assigning case to queue: ...")
         queue_id = self.context['queue_id'] if queue_id is None else queue_id
         case_id = self.context['case_id'] if case_id is None else case_id
         data = {'queues':  [queue_id]}
         self.make_request("PUT", url='/cases/' + case_id + '/', headers=self.gov_headers, body=data)
 
+    def assign_test_cases_to_bin(self, bin_queue_id):
+        self.log("assigning cases to bin: ...")
+        new_cases_queue_id = "00000000-0000-0000-0000-000000000001"
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/helpers/seed_data.py]: Probably should be defined in constants

File: ui_automation_tests/pages/users_page.py [hunk 2]
```diff
@@ -8,6 +8,9 @@ class UsersPage:
     def __init__(self, driver):
         self.driver = driver
 
+    def click_save_and_continue(self):
+        self.driver.find_element_by_css_selector("button[type*='submit']").click()
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/pages/users_page.py]: Maybe put these selectors in separate variables?

File: ui_automation_tests/fixtures/core.py [hunk 3]
```diff
@@ -4,6 +4,9 @@
 from selenium import webdriver
 from pytest import fixture
 
+from helpers.seed_data import SeedData
+from helpers.utils import get_or_create_attr
+
 
 def timeout_off(self):
     self.implicitly_wait(0)
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/fixtures/core.py]: I can see potential for tests never finishing if there's a problem. How about we give it a value of a minute instead?

File: ui_automation_tests/helpers/seed_data.py [hunk 4]
```diff
@@ -205,13 +205,31 @@ def add_queue(self, queue_name):
...
+        response = self.make_request("GET", url='/queues/' + new_cases_queue_id + '/', headers=self.gov_headers)
+        print(response)
+        queue = json.loads(response.text)['queue']
+        cases = queue['cases']
+        # cases = json.loads(queue.text)['cases']
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/helpers/seed_data.py]: What purpose does this comment serve?

File: ui_automation_tests/step_defs/test_flags.py [hunk 5]
```diff
@@ -88,8 +94,9 @@ def reactivate_first_deactivated_flag(driver):
 
 @then('I see the original number of active flags')
 def i_see_the_original_number_of_active_flags(driver, context):
-    number_of_active_flags = len(driver.find_elements_by_xpath('//*[text()[contains(.,"Active")]]'))
-    number_of_deactivated_flags = len(driver.find_elements_by_xpath('//*[text()[contains(.,"Deactivated")]]'))
+    table = driver.find_element_by_css_selector(".lite-table__body").text
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/step_defs/test_flags.py]: Should this kind of stuff not be extracted out to a "pages" file?

File: ui_automation_tests/conftest.py [hunk 6]
```diff
@@ -184,7 +156,7 @@ def create_clc_query(driver, apply_for_clc_query, context):
 def click_on_clc_case_previously_created(driver, context):
     case_list_page = CaseListPage(driver)
     assert case_list_page.assert_case_is_present(context.case_id)
-    driver.find_element_by_css_selector('.lite-cases-table').find_element_by_xpath("//*[text()[contains(.,'" + context.case_id + "')]]").click()
+    driver.find_element_by_css_selector('.lite-cases-table [href*="' + context.case_id + '"]').click()
```
Comment (markm5A8Z9F43913A) [inline ui_automation_tests/conftest.py]: Should this kind of stuff not be extracted out to a "pages" file?

File: ui_automation_tests/fixtures/apply_for_application.py [hunk 7]
```diff
@@ -49,182 +50,8 @@ def apply_for_standard_application(driver, request, context):
...
-    case_id = add_goods_page.assert_good_is_displayed_and_return_case_id(context.good_description)
-    context.case_id = case_id
-    driver.get(request.config.getoption("--internal_url"))
-
-
 @fixture(scope="module")
-def apply_for_clc_query(driver, request, context):
-    api = get_or_create_attr(context, 'api', lambda: SeedData(logging=True))
+def apply_for_clc_query(driver, request, api_url, context):
+    api = get_or_create_attr(context, 'api', lambda: SeedData(api_url=api_url, logging=True))
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/apply_for_application.py]: Nice Nice

File: ui_automation_tests/fixtures/core.py [hunk 8]
```diff
@@ -57,14 +60,20 @@ def sso_login_info(request):
...
 @fixture(scope="module")
 def invalid_username(request):
     return "invalid@mail.com"
+
+
+def clear_down(driver, context, api_url):
+    api = get_or_create_attr(context, 'api', lambda: SeedData(api_url=api_url, logging=True))
+    print(api.get_queues())
+    if "Application Bin" not in str(api.get_queues()):
+        api.add_queue("Application Bin")
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/core.py]: Niiice

File: ui_automation_tests/step_defs/test_flags.py [hunk 9]
```diff
@@ -61,20 +62,25 @@ def deactivate_first_active_flag(driver):
 
 @when('I click include deactivated')
 def click_include_deactivated(driver):
+    driver.timeout_off()
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_flags.py]: Does this mean that it'll `never` time-out? what happens if the page has failed to load for some reason - will the test spin forever?

File: ui_automation_tests/step_defs/test_flags.py [hunk 10]
```diff
@@ -61,20 +62,25 @@ def deactivate_first_active_flag(driver):
...
     if len(driver.find_element_by_css_selector("[href*='/flags/active/']")) == 1:
         driver.find_element_by_css_selector("[href*='/flags/active/']").click()
+    driver.timeout_on()
 
 
 @then('I see one less active flags')
 def i_see_one_less_active_flag(driver, context):
-    number_of_active_flags = len(driver.find_elements_by_xpath('//*[text()[contains(.,"Active")]]'))
-    number_of_deactivated_flags = len(driver.find_elements_by_xpath('//*[text()[contains(.,"Deactivated")]]'))
+    table = driver.find_element_by_css_selector(".lite-table__body").text
```
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_flags.py]: `.lite-table__body` into a variable pls

File: ui_automation_tests/step_defs/test_flags.py (hunk 5 above)
Comment (andytaylorkainos) [inline ui_automation_tests/step_defs/test_flags.py]: Again, variable pls

File: ui_automation_tests/fixtures/core.py (hunk 8 above)
Comment (andytaylorkainos) [reply ui_automation_tests/fixtures/core.py]: Now `might` be a good time to create a static `Application Test` queue so that we can always use this queue for testing.

It would also be worth removing the steps that create a new queue when testing `sort` and `filter` (instead, use the static `Application Test` queue)

All we'd need to do is ensure that the queue `Application Test` queue exists on start-up of the `session` - if it does exist, ensure it's empty (clear it if it's not empty) and if doesn't exist then create it.

File: ui_automation_tests/conftest.py [hunk 11]
```diff
@@ -23,7 +23,6 @@
 def pytest_exception_interact(node, report):
     if node and report.failed:
         class_name = node._nodeid.replace(".py::", "_class_")
-        name = "{0}_{1}".format(class_name, exporter_url)
```
Comment (andytaylorkainos) [inline ui_automation_tests/conftest.py]: I think we really should be trying to figure out how to get screenshots working again and refactoring the code before removing lines

File: ui_automation_tests/helpers/seed_data.py (hunk 1 above)
Comment (phlptt) [reply ui_automation_tests/helpers/seed_data.py]: created a new fixture file 'constants'

File: ui_automation_tests/pages/users_page.py (hunk 2 above)
Comment (phlptt) [reply ui_automation_tests/pages/users_page.py]: should do this, whole user page needs refactoring, i left this one when i moved all the code into this bdd format so it isnt done at all

File: ui_automation_tests/fixtures/core.py (hunk 3 above)
Comment (phlptt) [reply ui_automation_tests/fixtures/core.py]: sets it to 0 seconds for a timeout so if something isnt found within 0 seconds it times out basically

File: ui_automation_tests/helpers/seed_data.py (hunk 4 above)
Comment (phlptt) [reply ui_automation_tests/helpers/seed_data.py]: my mistake removed

File: ui_automation_tests/step_defs/test_flags.py (hunk 5 above)
Comment (phlptt) [reply ui_automation_tests/step_defs/test_flags.py]: this whole file needs refactoring, can i get this branched merged and in todays branch refactor this whole page please

File: ui_automation_tests/conftest.py (hunk 11 above)
Comment (phlptt) [reply ui_automation_tests/conftest.py]: looks like all this does is add the exporter_url and class_name in order to return some info when screenshot is saved, considering this is just returning a static exporter_url (which doesnt exist any more as have removed this dependency, ive just changed it to return class_name

File: ui_automation_tests/fixtures/constants.py [hunk 12]
```diff
@@ -0,0 +1,6 @@
+from pytest import fixture
```
Comment (andytaylorkainos) [inline ui_automation_tests/fixtures/constants.py]: could this fixture not be included in `fixtures/core.py`
